cachi2/core/package_managers/pip.py

```diff
@@ -1724,32 +1772,47 @@ def download_info(self) -> dict[str, Any]:
 
 def _process_package_distributions(
     requirement: PipRequirement, pip_deps_dir: RootedPath, allow_binary: bool = False
-) -> tuple[Optional[DistributionPackageInfo], list[DistributionPackageInfo]]:
+) -> list[DistributionPackageInfo]:
+    """
+    Return a DistributionPackageInfo object | a list of DPI objects, for the provided pip package.
+
+    Scrape the package's PyPI page and generate a list of all available
+    artifacts.
+    Filter by version and allowed artifact type.
+    Filter to find the best matching sdist artifact.
+    Process wheel artifacts.
+
+    :param requirement: which pip package to process
+    :param str pip_deps_dir:
+    :param bool allow_binary: process wheels?
+    :return: a single DistributionPackageInfo, or a list of DPI
+    :rtype: DistributionPackageInfo
+    """
+    allowed_distros = ["sdist", "wheel"] if allow_binary else ["sdist"]
+    artifacts: list[DistributionPackageInfo] = []
+    best_sdist: list[DistributionPackageInfo] = []
```

Comment:
due to the return statement, however, it can be tweaked.

Reply:
@eskultety;

> it can be tweaked

Explain?

Reply:
I want to return a single list from `_process_package_distributions`, no matter what.
The pypi type of the artifact is embedded in the DPI, so returning two lists is over-complicated.
Even treating `best_sdist` differently from everything else complicates the code unnecessarily (admittedly, it's a little weird to have it essentially be a single-element list, but not so mind-blowing as to confuse anyone).

Reply:
>>    it can be tweaked

> Explain?

That was a forward reference to my later comment here: https://github.com/containerbuildsystem/cachi2/pull/467#discussion_r1497943711